mod.go

```diff
@@ -0,0 +1,624 @@
+// uint256: Fixed size 256-bit math library
+// Copyright 2021 uint256 Authors
+// SPDX-License-Identifier: BSD-3-Clause
+
+package uint256
+
+import (
+	"math/bits"
+	"sync"
+)
+
+// Cache for reciprocal()
+//
+// Each cache set contains its own mutex to reduce lock contention in highly
+// multithreaded settings.
+//
+// Adjust cacheIndexBits and cacheWays to scale the size of the cache.
+// Total cache size is (24+72*cacheWays)*2^cacheIndexBits bytes, which is
+// 96 KiB with cacheIndexBits = 8 and cacheWays = 5. There are also 16 bytes
+// for hit and miss counters.
+//
+// Reasonable values are quite small, e.g. cacheIndexBits from 2 to 10, and
+// cacheWays around 5. Note that cacheWays = 5+8n makes the set size an integer
+// number of 64-byte cachelines.
+//
+// If you want to disable the cache, set cacheIndexBits and cacheWays to 0
```

Comment:
This doesn't really work, at least not if the 'consumer' of the library wants to use it as a go module. The only way to make this work is to copy-paste the entire library and modify the value, since it's a `const`.

Reply:
The constness allows the compiler to allocate the cache memory statically and to remove unused code paths. Does Go have a construct that can replace templates, allowing users to instantiate the cache with their own parameters? Or maybe the cache needs to be dynamically allocated to give users this flexibility, at the risk of losing performance?